### PR TITLE
Lvnod 77 create newevent component

### PR DESCRIPTION
**Pull request by yarvanya**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-77)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-106)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-107)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-108)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-125)

## Code reviewers

- [ ] @yurii-kokhalevych
- [ ] @denisbodnar
- [ ] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

- First, I have created newEvent container component.
- Second, I've added route for this.
- Third, I have created the newEventReducer with all needed actions for the newEvent container component.
- Fourth, I've added my child newEventReducer for the main reducer.
- Fifth, I have changed simple date_event field with dateTimePicker and added some styles for this.

## CHECK LIST
- [x]  СI passed
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/router.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const Routes = (
     </Route>
     <Route path='signup' component={Signup}/>
     <Route path='events' component={EventsList}/>
+    <Route  path='newevent' component={Newevent}/>
```
Comment (4asof4ik) [inline app/src/router.js]: camelCase

File: app/src/redux/reducer.js [hunk 2]
```diff
@@ -4,11 +4,13 @@ import signup from './Signup';
 import login from './Login';
 import event from './eventReducers';
 import eventsList from './EventsReducers';
+import newevent from './createNewEventReducer';
```
Comment (4asof4ik) [inline app/src/redux/reducer.js]: without create and naming files shouldn't be different, fix this

File: app/src/containers/Newevent/Newevent.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        name: null,
+        date_event: null
+      },
+      enableButton: false
+    };
+  }
+
+  getValidationState = fieldName => {
+    const pattern = {
+      name: /^.{4,}$/,
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: drop this and check if event length more than 4

File: app/src/containers/Newevent/Newevent.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      date_event:  /^(0?[1-9]|[12][0-9]|3[01])[/](0?[1-9]|1[012])[/]\d{4}$/
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: drop and use datapicker

File: app/src/containers/index.js [hunk 5]
```diff
@@ -9,3 +9,4 @@ export EventDetails from './Event/EventDetails';
 export Guest from './Guest/sampleGuest';
 export Gift from './Gift/sampleGift';
 export Signup from './Signup/Signup';
+export Newevent from './Newevent/Newevent';
```
Comment (4asof4ik) [inline app/src/containers/index.js]: camelCase

File: app/src/containers/Newevent/Newevent.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  handleChange = param => e => {
+    let value = e.target.value.trim();
+    this.setState({ [param]: value},
+      () => {this.getValidationState(param);});
+  };
+
+  handleButtonClick = e => {
+    e.preventDefault();
+    this.props.actions.createNewEvent(
+      this.state.name,
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: why you send each state

File: app/src/redux/createNewEventReducer.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  };
+}
+
+export function createNewEventRequest() {
+  return {
+    type: CREATE_NEW_EVENT_REQUEST
+  };
+}
+
+export function createNewEvent(name, date_event, location_name, longitude, latitude, description) {
```
Comment (4asof4ik) [inline app/src/redux/createNewEventReducer.js]: O_o

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/router.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { Route, IndexRoute } from 'react-router';
 import { About, Address, Home, Contacts, Main, Maps, Signup,
-  Gift, EventsList, EventDetails, Profile } from './containers';
+  Gift, Guest, EventsList, EventDetails, Profile, newEvent } from './containers';
```
Comment (4asof4ik) [inline app/src/router.js]: Guest?

File: app/src/redux/newEventReducers.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import request from '../helper/request';
+import { API } from './../helper/constants';
+import { push } from 'react-router-redux';
+import { browserHistory } from 'react-router';
+const CREATE_NEW_EVENT_SUCCESS = 'CREATE_NEW_EVENT_SUCCESS';
```
Comment (4asof4ik) [inline app/src/redux/newEventReducers.js]: Empty line

File: app/src/redux/newEventReducers.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  browserHistory.push('/events');
+  return {
+    type: CREATE_NEW_EVENT_SUCCESS
+  };
+}
+
+export function createNewEventFailure() {
+  return {
+    type: CREATE_NEW_EVENT_FAILURE
+  };
```
Comment (4asof4ik) [inline app/src/redux/newEventReducers.js]: Where error

File: app/src/redux/newEventReducers.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      };
+    default:
+      return state;
+  }
+}
+
+export function createNewEventSuccess() {
+  browserHistory.push('/events');
+  return {
+    type: CREATE_NEW_EVENT_SUCCESS
```
Comment (4asof4ik) [inline app/src/redux/newEventReducers.js]: Where event

File: app/src/redux/newEventReducers.js (hunk 10 above)
Comment (4asof4ik) [inline app/src/redux/newEventReducers.js]: Where error

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/redux/reducer.js [hunk 12]
```diff
@@ -6,6 +6,7 @@ import event from './eventReducers';
 import profile from './profileReducers';
 import eventsList from './eventsReducers';
 import invite from './invite';
```
Comment (4asof4ik) [inline app/src/redux/reducer.js]: rename inviteReducers

File: app/src/containers/Newevent/Newevent.js [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+    } else {
+      newState.isErrors.name = null;
+      newState.enableButton = true; 
+    }
+    this.setState(newState);
+  };
+
+  handleChange = param => e => {
+    const value = e.target.value.trim();
+    this.setState({ [param]: value}, () => {this.getValidationState();});
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: drop space { [

File: app/src/containers/Newevent/Newevent.js [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+      latitude: 'Latitude',
+      description: 'Descripton'
+    };
+    return (
+      <Col sm={9}>
+        <h2>Here, you can create your own event:</h2>
+        <Form onSubmit={this.handleButtonClick}>
+          { Object.keys(inputsEventData).map(param =>
+            (param == 'date_event') ?
+              <FormGroup key={param}>
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: create const and use them 99 line too

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: app/src/containers/Newevent/Newevent.js [hunk 15]
```diff
@@ -0,0 +1,150 @@
+import  React from 'react';
+import { FormGroup, ControlLabel, FormControl, Button, Form, Col, HelpBlock, Label } from 'react-bootstrap';
+import { browserHistory } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as showActions from '../../redux/newEventReducers';
+import messages from '../../helper/messages';
+import DateTimeField from 'react-bootstrap-datetimepicker';
```
Comment (4asof4ik) [inline app/src/containers/Newevent/Newevent.js]: first  include all vendor after that own

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**